### ISSUE TITLE
printing in ocamldebug

### ISSUE THREAD

**Issue by nilsbecker** [mmottl/lacaml#24]
is there a way of installing lacaml's vector and matrix printers under ocamldebug?

 i looked for .cma or .cmo files in the distribution and found only lacaml.cma and lacaml_top.cma. the Lacaml_io module in particular does not have a bytecode object. i was unable to load lacaml{,_top}.cma in the debugger due to missing function definitions. am i doing it wrong? printing in the toploop works just fine.

**Comment by mmottl** [comment]
Yes, you can install Lacaml printers.  You'll probaby have to enter something like the following or put it in the `.ocamldebug` init file in the directory of your project:

```
directory ~/.opam/4.04.0/lib/lacaml

load_printer bigarray.cma
load_printer lacaml.cma

install_printer Lacaml_io.pp_fmat
```

Once you have reached a break point, you can then enter something like `d mat` to display a double float matrix.  You may want to install more printers for vectors, other element types, etc.

**Comment by nilsbecker** [comment]
thanks! the part i was missing was just the fact that bigarray.cma has to be loaded first. that's not obvious from the error message one gets.

**Comment by nilsbecker** [comment]
update, the function names are now more like `Lacamla__io.pp_fmat`  with two underscores.

**Comment by mmottl** [comment]
`Lacaml.Io.pp_fmat` should be the right module path.  My fault, I should have used this instead of `Lacaml_io` in my previous solution.

**Comment by nilsbecker** [comment]
actually, i got some segfaults from ocamldebug when i tried `Lacaml.Io.some_printer`. not sure what printing function i tried exactly; i'll verify if this is the case for `pp_fmat`

**Comment by nilsbecker** [comment]
yes, i still get a segfault when using `Lacaml.Io.pp_fmat` in `.ocamldebug` instead of the underscored module path.

**Comment by mmottl** [comment]
I've replicated this problem, which may indicate a bug in the OCaml runtime.  Here is a backtrace:

```
* thread #1, stop reason = signal SIGSTOP
  * frame #0: 0x0000000106973a00 ocamlrun`caml_array_unsafe_get
    frame #1: 0x000000010696424b ocamlrun`caml_interprete + 7931
    frame #2: 0x0000000106966f75 ocamlrun`caml_main + 1557
    frame #3: 0x000000010698507c ocamlrun`main + 12
    frame #4: 0x00007fffb7782235 libdyld.dylib`start + 1
    frame #5: 0x00007fffb7782235 libdyld.dylib`start + 1
```

The above function works just fine when executed via an ordinary executable.  Not sure why it doesn't work when installed in the debugger with the above module path.

I don't have time to investigate this further right now.  Please feel free to dig deeper into this, maybe you can file a bug report with the OCaml team.

**Comment by nilsbecker** [comment]
it seems there is an issue about this already, which is fixed in 4.05. 
https://caml.inria.fr/mantis/view.php?id=7258

**Comment by mmottl** [comment]
Thanks for finding this Mantis issue, which seems to be the exact same problem.
